doc/adding_to_an_adapter.md

```diff
@@ -1,120 +1,237 @@
+> ```
+>> Note: The `AdapterDefinition` will generate a describe.xml equivalent to the one below
+> ### describe.xml
+> ```xml
+> <AdapterKind key="MyAdapter" nameKey="My Adapter" version="1" >
+>     <CredentialKinds/>
+>     <ResourceKinds>
+>         <ResourceKind key="MyAdapter_adapter_instance" nameKey="2" type="7" credentialKind=""/>
+>     </ResourceKinds>
+> </AdapterKind>
+> ```
 
-```xml
-<AdapterKind xmlns="http://schemas.vmware.com/vcops/schema" key="my_adapter" nameKey="1" version="1">
-    <CredentialKinds/>
-    <ResourceKinds/>
-    <!-- optional CapacityDefinitions, TraversalSpecKinds, etc -->
-</AdapterKind>
-```
 For more information about the supported elements and attributes, see the [describe.xml documentation](../vmware_aria_operations_integration_sdk/adapter_template/describeSchema.xsd).
+For more information about the Python module visit [VMware Aria Operations Integration SDK Library](https://pypi.org/project/vmware-aria-operations-integration-sdk-lib/).
 
-## Defining an Adapter Instance
-An adapter instance is a special object in VMware Aria Operations that stores user configuration for a connection. Every adapter must have exactly one adapter instance. Adapter instances are set by defining a `ResourceKind` element with attribute `type=7`.
-```xml
-<AdapterKind xmlns="http://schemas.vmware.com/vcops/schema" key="my_adapter" nameKey="1" version="1">
-  <ResourceKinds>
-    <ResourceKind key="my_adapter_instance" nameKey="4" type="7" >
-        <!-- ... -->
-    </ResourceKind>
-  </ResourceKinds>
-</AdapterKind>
-```
-
-Once an adapter instance is defined, any configuration fields (`ResourceIdentifiers`) and credentials (`CredentialKind`) will be prompted to the user when creating an account in VMware Aria Operations on the `Data Sources` &rarr; `Integrations` page (See [Adding a Configuration Field to an Adapter Instance](#adding-a-configuration-field-to-an-adapter-instance) and [Adding a Credential](#adding-a-credential).
-After the account has been created, configuration fields will be available in the input to the `collect`, `test_connection`, and `get_endpoint_urls` methods. (See [Creating an Adapter Instance](#creating-an-adapter-instance).)
+Once an adapter instance is defined, any configuration fields (`ResourceIdentifiers` element) and credentials (`CredentialKind` element)
+will be prompted to the user when creating an account in VMware Aria Operations on the `Data Sources` &rarr; `Integrations`
+page (See [Adding a Configuration Field to an Adapter Instance](#adding-a-configuration-field-to-an-adapter-instance)
+and [Adding a Credential](#adding-a-credential). After the account has been created, configuration fields will be
+available in the input to the `collect`, `test`, and `get_endpoint` methods.
+(See [Creating an Adapter Instance](#creating-an-adapter-instance).)
 
-## Adding a Configuration Field to an Adapter Instance
+## Adding a Configuration Field to an Adapter Instance in the Object Model
```

Comment:
Any time you change a header like this, we need to make sure all of the anchor links are updated, otherwise they will no longer work. For example, on line 51 we have this line:

`
[Adding a Configuration Field to an Adapter Instance](#adding-a-configuration-field-to-an-adapter-instance)
`

which links to the old version of this header. It will have to be changed to this:

`
[Adding a Configuration Field to an Adapter Instance](#adding-a-configuration-field-to-an-adapter-instance-in-the-object-model)
`

This will have to be done for all of the header updates. Some markdown plugins will detect broken links and help you fix them, I'd recommend making sure you have one that does and the feature is turned on.